### PR TITLE
envoy: Use embedded proxylib from cilium-proxy image

### PR DESCRIPTION
**Pull request by sayboras**
### Description

The goal is to address one known limitation of the dedicated envoy proxy mode. Please refer to individual commit for more details.

### Testing

Testing was done with the existing conformance-ginkgo job, the kafka-related policy tests are passed as per below:

- https://github.com/cilium/cilium/actions/runs/5355371399/jobs/9713634668
- https://github.com/cilium/cilium/actions/runs/5355371399/jobs/9713636374
- https://github.com/cilium/cilium/actions/runs/5355371399/jobs/9713638899

### CODE REVIEW

**Comment by mhofstetter** [comment]
@sayboras as mentioned, it's possible to remove the currently documented limitation in the docs https://github.com/cilium/cilium/blob/main/Documentation/security/network/proxy/envoy.rst?plain=1#L61 :tada:

**Comment by jrajahalme** [comment]
Now that https://github.com/cilium/proxy/pull/232 has merged we can update this to newest main references :-)

**Comment by sayboras** [comment]
> Now that [cilium/proxy#232](https://github.com/cilium/proxy/pull/232) has merged we can update this to newest main references :-)

Thanks, let me make an update shortly :+1:

**Review by rolinh — APPROVED** [review]
Vendor changes lgtm

**Review by bimmlerd — COMMENTED** [review]
CODEOWNERS change lgtm (except that I fatfingered the wrong radio button, lol)

**Review by bimmlerd — APPROVED** [review]

**Review by learnitall — APPROVED** [review]
LGTM on docs!

File: CODEOWNERS
Comment (joestringer) [inline CODEOWNERS]: nit: I would think we can still use team @cilium/proxy for review on https://github.com/cilium/proxy, no?

File: proxylib/Makefile
Comment (joestringer) [inline proxylib/Makefile:1]: Do we still have proxylib documentation that needs to be updated for this change? If so, please fix that up as well. Doesn't need to block this PR.

**Review by joestringer — APPROVED** [review]
Minor nits, seems like a straightforward migration to me.

File: CODEOWNERS
Comment (sayboras) [reply CODEOWNERS]: oh right, I was under the impression that codeowner is managed per repo. 

Let me revert this update.

File: proxylib/Makefile
Comment (sayboras) [reply proxylib/Makefile:1]: noted and thanks, I will send the follow-up PR on docs 👍

**Review by gandro — APPROVED** [review]
Looks good from my code owners

**Review by mhofstetter — APPROVED** [review]
LGTM - thanks Tam (also for removing the limitation)!

**Review by youngnick — APPROVED** [review]

**Review by nathanjsweet — APPROVED** [review]
LGTM.

**Pull request merged**
